torchgeo/datamodules/inria.py

```diff
@@ -39,214 +24,73 @@ class InriaAerialImageLabelingDataModule(pl.LightningDataModule):
-        return sample
-
-    def preprocess(self, sample: Dict[str, Any]) -> Dict[str, Any]:
-        """Transform a single sample from the Dataset.
-
-        Args:
-            sample: input image dictionary
-
-        Returns:
-            preprocessed sample
-        """
-        sample["image"] = sample["image"].float()
-        sample["image"] /= 255.0
-        sample["image"] = torch.clip(sample["image"], min=0.0, max=1.0)
-
-        if "mask" in sample:
-            sample["mask"] = rearrange(sample["mask"], "h w -> () h w")
-
-        return sample
-
-    def n_random_crop(self, sample: Dict[str, Any]) -> Dict[str, Any]:
-        """Get n random crops."""
-        images, masks = [], []
-        for _ in range(self.num_patches_per_tile):
-            image, mask = sample["image"], sample["mask"]
-            # RandomCrop needs image and mask to be in float
-            mask = mask.to(torch.float)
-            image, mask = self.random_crop(image, mask)
-            images.append(image.squeeze())
-            masks.append(mask.squeeze(0).long())
-        sample["image"] = torch.stack(images)  # (t,c,h,w)
-        sample["mask"] = torch.stack(masks)  # (t, 1, h, w)
-        return sample
-
-    def setup(self, stage: Optional[str] = None) -> None:
-        """Initialize the main ``Dataset`` objects.
-
-        This method is called once per GPU per run.
-        """
-        train_transforms = T.Compose([self.preprocess, self.n_random_crop])
-        test_transforms = T.Compose([self.preprocess, self.patch_sample])
-
-        self.dataset = InriaAerialImageLabeling(
-            split="train", transforms=train_transforms, **self.kwargs
+            _RandomNCrop(self.patch_size, self.num_patches_per_tile),
+            data_keys=["image", "mask"],
         )
-
-        self.train_dataset: Dataset[Any]
-        self.val_dataset: Dataset[Any]
-        self.test_dataset: Dataset[Any]
-
-        if self.val_split_pct > 0.0:
-            if self.test_split_pct > 0.0:
-                self.train_dataset, self.val_dataset, self.test_dataset = dataset_split(
-                    self.dataset,
-                    val_pct=self.val_split_pct,
-                    test_pct=self.test_split_pct,
-                )
-            else:
-                self.train_dataset, self.val_dataset = dataset_split(
-                    self.dataset, val_pct=self.val_split_pct
-                )
-                self.test_dataset = self.val_dataset
-        else:
-            self.train_dataset = self.dataset
-            self.val_dataset = self.dataset
-            self.test_dataset = self.dataset
-
-        self.predict_dataset = InriaAerialImageLabeling(
-            split="test", transforms=test_transforms, **self.kwargs
+        self.val_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            _ExtractTensorPatches(self.patch_size),
+            data_keys=["image", "mask"],
```

Comment:
Think `val_aug` should be the same as `train_aug` just without the random rotations and flips.

Reply:
So you want to replace _ExtractTensorPatches with _RandomNCrop during validation? What about for the other datamodules that use this?

Reply:
It should be the same treatment. `_ExtractTensorPatches` is when we want all the patches to be considered i.e. during prediction. Otherwise the batch_size won't be honored and you would trigger OOM during validation in between training.

Reply:
Hmm, when evaluating (validation, test), don't we want to evaluate performance on the entire file, not just a few random patches?

Reply:
Hmm not sure. The validation/test scores are an indicator of how well the model has generalized the patterns seen in the training set on the validation/test sets. So more patches, the more representative the score is. I guess the issue is more practical. 

Assuming a patch_size of 512, each image (5000 x 5000) in the inria dataset can be considered as a (100, 3, 512, 512) tensor

During evaluation, 
If we use RandomNCrops, we're looking at (N, 3, 512, 512) images per iteration whereas with ExtractTensorPatches we're looking at (100, 3, 512, 512) images per iteration.

Reply:
Is your point that N != 100 or that 100 is too large? This is the behavior we decided on in #876. All data modules with large images do this:

* DeepGlobe
* GID-15
* Inria
* OSCD
* Potsdam
* Vaihingen

The data module docs do make things clear at least:
```
The Inria Aerial Image Labeling dataset contains images that are too large to    
pass directly through a model. Instead, we randomly sample patches from image    
tiles during training and chop up image tiles into patch grids during            
evaluation. During training, the effective batch size is equal to                
``num_tiles_per_batch`` x ``num_patches_per_tile``.        
```

Reply:
100 being too large. But I guess this might be hardware dependent

Reply:
We could define NonGeoSamplers so that we can emulate GridGeoSampler for NonGeoDatasets, but that feels like overkill. I would like to target the lowest possible hardware requirements though, so if it's a bottleneck we can change it. Are you able to run `trainer.predict` with `ExtractTensorPatches` though?

Reply:
But `predict_aug` also uses `ExtractTensorPatches`, so shouldn't it have the same problem? Are you saying you don't have this problem because you don't use `trainer.predict()`, you do things manually? We should get the former working if it isn't working.

Reply:
Yeah I didn't use `trainer.predict` in my prediction PR. I was manually running inference on each patch.

I'll take a look at creating predictions for inria using `trainer.predict` and see if it works.